### ISSUE TITLE
Verifier rejected class with HttpClient.get()

### ISSUE THREAD

**Issue by brewin** [ktorio/ktor#979]
### Ktor Version
1.1.3
### Ktor Engine Used(client or server and name)
Android client
### JVM Version, Operating System and Relevant Context
Tested on an Android 8 device and an Android 9 emulator
### Feedback
I am getting the following error when using `HttpClient.get()` in a suspend function that takes two non-nullable arguments. The error does not occur when the arguments aren't used in the body of the function or when one argument is nullable. It's hard to explain, so here is a [demo](https://github.com/brewin/verifier-error/blob/master/app/src/main/java/com/github/brewin/test/MainActivity.kt).

>     java.lang.VerifyError: Verifier rejected class com.github.brewin.test.MainActivity: java.lang.Object com.github.brewin.test.MainActivity.test(double, double, kotlin.coroutines.Continuation) failed to verify: java.lang.Object com.github.brewin.test.MainActivity.test(double, double, kotlin.coroutines.Continuation): [0x201] copy2 v8<-v38 type=Zero/null/Double (High Half) (declaration of 'com.github.brewin.test.MainActivity' appears in /data/app/com.github.brewin.test-Ix2r2ZQPTSpO3hi1mfdfSA==/base.apk)
>         at java.lang.Class.newInstance(Native Method)
>         at android.app.AppComponentFactory.instantiateActivity(AppComponentFactory.java:69)
>         at androidx.core.app.CoreComponentFactory.instantiateActivity(CoreComponentFactory.java:43)
>         at android.app.Instrumentation.newActivity(Instrumentation.java:1215)
>         at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2831)
>         at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3048)
>         at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:78)
>         at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108)
>         at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68)
>         at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1808)
>         at android.os.Handler.dispatchMessage(Handler.java:106)
>         at android.os.Looper.loop(Looper.java:193)
>         at android.app.ActivityThread.main(ActivityThread.java:6669)
>         at java.lang.reflect.Method.invoke(Native Method)
>         at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
>         at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)

**Comment by brewin** [comment]
Adding this to gradle.properties makes the error go away, so it seems to be related to D8.
```
android.enableD8=false
android.enableD8.desugaring=false
```

**Comment by e5l** [comment]
Hi @brewin, thanks for the report. I'll try to investigate the problem, it looks like the `D8` issue.

**Comment by cy6erGn0m** [comment]
The most simplified code is 

```kotlin

import android.os.*
import androidx.appcompat.app.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    private suspend fun test(d1: Double, d2: Double) {
        val test = d1

        val x = 1
        val y = 2
        val z = 4
        val t = 5

        Receiver.tier1()
    }
}

object Receiver

suspend fun receiveTopLevel(): Unit = TODO()
suspend fun executeTopLevel(p: Any): Unit = TODO()

suspend inline fun Receiver.tier1() {
    tier2("")
}

suspend inline fun Receiver.tier2(
    something: String
) {
    executeTopLevel(something)
    tier3()
}

suspend inline fun Receiver.tier3() {
    typeInfo2()
    receiveTopLevel()
}

inline fun typeInfo2() {
    (f() as List<Any?>).first()!!
}

fun f(): Any {
    TODO()
}
```

**Comment by cy6erGn0m** [comment]
This code above causes dex verifier to crash at [CopyRegister2](https://android.googlesource.com/platform/art/+/26fee74/src/dex_verifier.cc#673) with the following error mesage:

> [0x114] copy2 v9<-v25 type=Zero/null/Double (High Half) 

DEX disassembed code is the following:

```dex
  148:   0xfd: InvokeStatic        { v12 v1 } 
                                           Ltest/MainActivityKt;->executeTopLevel(
                                                Ljava/lang/Object;
                                                Lkotlin/coroutines/Continuation;
                                           )Ljava/lang/Object;
  149:  0x100: MoveResultObject    v0
  150:  0x101: IfNe                v0, v3, 0x104 (+3)
         0x103, line 13, locals: [1 -> $continuation, 2 -> x, 4 -> test, 
                                           8 -> z, 9 -> t, 10 -> $receiver$iv, 
                                           11 -> $i$f$tier1, 12 -> something$iv$iv, 
                                           13 -> $receiver$iv$iv,
                                           14 -> $i$f$tier2, 15 -> y, 24 -> this,
                                           25 -> d1, 27 -> d2]
  151:  0x103: ReturnObject        v3
         0x104, line 61, locals: [1 -> $continuation, 2 -> x, 4 -> test, 
                                           8 -> z, 9 -> t, 10 -> $receiver$iv, 
                                           11 -> $i$f$tier1, 12 -> something$iv$iv, 
                                           13 -> $receiver$iv$iv, 14 -> $i$f$tier2, 
                                           15 -> y, 24 -> this, 
                                           25 -> d1, 27 -> d2]
  152:  0x104: MoveFrom16          v25, v11   // copy int to double
  153:  0x106: MoveObjectFrom16    v16, v12
  154:  0x108: Move                v0, v14
  155:  0x109: Move                v14, v15
  156:  0x10a: MoveObject          v15, v3
  157:  0x10b: MoveWide            v11, v4
  158:  0x10c: MoveObject          v5, v13
  159:  0x10d: MoveObjectFrom16    v4, v24
  160:  0x10f: Move                v13, v8
  161:  0x110: Move                v8, v2
  162:  0x111: MoveWide            v2, v6
  163:  0x112: Move                v7, v9
  164:  0x113: MoveObject          v6, v10
  165:  0x114: MoveWideFrom16      v9, v25  // verify error is here
```

**Comment by cy6erGn0m** [comment]
Opened an issue against D8 https://issuetracker.google.com/issues/126273993

**Comment by cy6erGn0m** [comment]
Fixed in R8 1.4.60. See instructions in the issue linked above to get the fixed early.
